crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -126,7 +126,7 @@ impl SystemMeta {
 /// world.resource_scope(|world, mut cached_state: Mut<CachedSystemState>| {
 ///     let mut event_reader = cached_state.event_state.get_mut(world);
 ///
-///     for events in event_reader.iter(){
+///     for events in event_reader.iter() {
```

Comment:
Is it intentional that there is still an `iter()` call here?

Reply:
I also noticed this, but `event_reader`'s type does not implement `IntoIterator`, so you can't omit this `.iter()`.

Reply:
Indeed. Ideally all of our types that have `.iter()` methods would implement `IntoIterator` which *is* possible but there are some unfortunate sacrifices that would need to be made.
There's a conversation about this in a 'resolved' comment on [this](https://github.com/bevyengine/bevy/pull/5096) pr.
If [TAIT](https://github.com/rust-lang/rfcs/pull/2515) was stabilized howerver, we could impl `IntoIterator` with ease.